code/modules/power/supermatter/supermatter.dm

```diff
@@ -1,86 +1,157 @@
-	var/explosion_point = 900
+	var/explosion_point = 1000
 
-	var/emergency_issued = 0
+	var/warning_color = "#B8B800"
+	var/emergency_color = "#D9D900"
 
-	var/explosion_power = 8
+	var/grav_pulling = 0
+	var/pull_radius = 6
+	// Time in ticks between delamination ('exploding') and exploding (as in the actual boom)
+	var/pull_time = 400
+	var/pull_ticks = 4
+	var/explosion_power = 6
 
-	var/lastwarning = 0				// Time in 1/10th of seconds since the last sent warning
-	var/power = 0
+	var/emergency_issued = 0
+
+	// Time in 1/10th of seconds since the last sent warning
+	var/lastwarning = 0
 
-	var/oxygen = 0					// Moving this up here for easier debugging.
+	// This stops spawning redundand explosions. Also incidentally makes supermatter unexplodable if set to 1.
+	var/exploded = 0
 
+	var/power = 0
+	var/oxygen = 0
+	var/sound_played = 0
 	//Temporary values so that we can optimize this
 	//How much the bullets damage should be multiplied by when it is added to the internal variables
 	var/config_bullet_energy = 2
 	//How much of the power is left after processing is finished?
-//	var/config_power_reduction_per_tick = 0.5
+//        var/config_power_reduction_per_tick = 0.5
 	//How much hallucination should it produce per unit of power?
-	var/config_hallucination_power = 0.1
+	var/config_hallucination_power = 0.02
 
 	var/obj/item/device/radio/radio
 
-	//for logging
+	var/debug = 0
 	var/has_been_powered = 0
-	var/has_reached_emergency = 0
 
-/obj/machinery/power/supermatter_shard/New()
+/obj/machinery/power/supermatter/New()
 	. = ..()
-	radio = new(src)
-	radio.listening = 0
-	investigate_log("has been created.", "supermatter")
+	radio = new (src)
 
 
-/obj/machinery/power/supermatter_shard/Destroy()
-	investigate_log("has been destroyed.", "supermatter")
-	qdel(radio)
+/obj/machinery/power/supermatter/Del()
+	del radio
 	. = ..()
 
-/obj/machinery/power/supermatter_shard/proc/explode()
-	investigate_log("has exploded.", "supermatter")
-	explosion(get_turf(src), explosion_power, explosion_power * 2, explosion_power * 3, explosion_power * 4, 1)
-	qdel(src)
-	return
+/obj/machinery/power/supermatter/proc/explode()
+	message_admins("Supermatter exploded at ([x],[y],[z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>)",0,1)
+	log_game("Supermatter exploded at ([x],[y],[z])")
+	anchored = 1
+	grav_pulling = 1
+	exploded = 1
+	for(var/mob/living/mob in living_mob_list)
+		if(loc.z == mob.loc.z)
+			if(istype(mob, /mob/living/carbon/human))
+				//Hilariously enough, running into a closet should make you get hit the hardest.
+				var/mob/living/carbon/human/H = mob
+				H.hallucination += max(0, min(300, DETONATION_HALLUCINATION * sqrt(1 / (get_dist(mob, src) + 1)) ) )
+			var/rads = DETONATION_RADS * sqrt( 1 / (get_dist(mob, src) + 1) )
+			mob.apply_effect(rads, IRRADIATE)
+	spawn(pull_time)
+		explosion(get_turf(src), explosion_power, explosion_power * 2, explosion_power * 4, explosion_power * 6, 1, 1)
+		del src
```

Comment:
>del
where did you copy this code from?some very very old source? edit this to use qdel